battleship.py

```diff
@@ -145,7 +244,14 @@ def drawShip(data, canvas, ship):
 Returns: bool
 '''
 def shipIsValid(grid, ship):
-    return
+   if(checkShip(grid,ship)==True&(isVertical(ship)==True|isHorizontal(ship)==True)):
+        return True
+   else:
+        return False

```

Comment:
1. Use 'and' and 'or' operators instead of '&' and '|' as they are bitwise operators.
2. check length of ship also
3. else is not needed, you can directly return false.
4. no need of check equality with true for functions which return boolean